core/common/src/main/java/alluxio/AbstractClient.java

```diff
@@ -195,11 +195,10 @@ public synchronized void connect() throws IOException, ConnectionFailedException
    */
   public synchronized void disconnect() {
     if (mConnected) {
+      Preconditions.checkNotNull(mProtocol);
```

Comment:
Preconditions.checkNotNull(mProtocol, "mProtocol should never be null when mConnected is true");